web/src/main/java/org/mskcc/cbio/portal/web/ProxySessionServiceController.java

```diff
@@ -0,0 +1,219 @@
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.client.RestTemplate;
+
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+@Controller
+@RequestMapping("/proxy/session")
+public class ProxySessionServiceController {
+
+    @Value("${session.service.url:''}")
+    private String sessionServiceURL;
+    
+    
+    @RequestMapping(value = "/{type}/{id}", method = RequestMethod.GET)
+    public @ResponseBody Object getSessionService(
+            @PathVariable SessionType type, 
+            @PathVariable String      id,
+            HttpServletResponse       response) throws URISyntaxException, IOException {
+        
+        try {
+            RestTemplate restTemplate = new RestTemplate();
+
+            ResponseEntity<HashMap> responseEntity = restTemplate.exchange(sessionServiceURL + type + "/" + id,
+                                                                           HttpMethod.GET,
+                                                                           null,
+                                                                           HashMap.class);
+            
+            if(type.equals(SessionType.virtual_study)) {
+                ObjectMapper mapper = new ObjectMapper()
+                		.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+                
+                String virtualStudyStr = mapper.writeValueAsString(responseEntity.getBody());
+                
+                return mapper.readValue(virtualStudyStr, VirtualStudy.class);
+                
+            }
+            return responseEntity.getBody();
+        } catch (Exception exception) {
+        	exception.printStackTrace();
+            String errorMessage = "Unexpected error: " + exception.getMessage();
+            response.sendError(503, errorMessage);
+        }
+        return null;
+    }
+    
+    @RequestMapping(value = "/virtual_study", method = RequestMethod.GET)
+    public @ResponseBody List<VirtualStudy> getUserStudies() {
+    	
+        List<VirtualStudy> virtualStudies = new ArrayList<>();
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+        if (authentication != null && !(authentication instanceof AnonymousAuthenticationToken)) {
+            RestTemplate restTemplate = new RestTemplate();
+            ResponseEntity<VirtualStudy[]> responseEntity = restTemplate.getForEntity(
+                    sessionServiceURL + "virtual_study/query?field=data.users&value=" + authentication.getName(),
```

Comment:
Did we decide to use name?

Reply:
`getName()` in `Authentication` refers to `userName` for social authentication. For clarification will get it from principal.

Reply:
Just curious, wouldn't be easier to use the email address to identify users?